src/pages/outlier-detection/OutlierDetection.js

```diff
@@ -262,53 +262,53 @@ class OutlierDetection extends Page {
-    //                 apiConf.endpoints.markDataValue,
-    //                 OutlierAnalyisTable.convertElementToToggleFollowupRequest(
-    //                     currentElement
-    //                 )
-    //             )
-    //                 .then(() => {
-    //                     if (this.isPageMounted()) {
-    //                         currentElement.marked = !currentElement.marked
-    //                         elements[i] = currentElement
-
-    //                         this.context.updateAppState({
-    //                             showSnackbar: true,
-    //                             snackbarConf: {
-    //                                 type: SUCCESS,
-    //                                 message: i18n.t(
-    //                                     currentElement.marked
-    //                                         ? 'Marked'
-    //                                         : 'Unmarked'
-    //                                 ),
-    //                             },
-    //                             pageState: {
-    //                                 elements,
-    //                                 loading: false,
-    //                             },
-    //                         })
-    //                     }
-    //                 })
-    //                 .catch(error => {
-    //                     this.manageError(error)
-    //                 })
-    //             break
-    //         }
-    //     }
-    // }
+    toggleCheckbox(element) {
+        const api = this.context.d2.Api.getApi()
+        const elements = this.state.elements
+        for (let i = 0; i < elements.length; i++) {
+            const currentElement = elements[i]
+            if (currentElement.key === element.key) {
+                this.context.updateAppState({
+                    pageState: {
+                        loading: true,
+                        elements,
+                        showTable: elements && elements.length > 0,
+                    },
+                })
+                const data = OutlierAnalyisTable.convertElementToToggleFollowupRequest(
+                    currentElement
+                )
+                api.update(apiConf.endpoints.markDataValue, data)
+                    .then(() => {
+                        if (this.isPageMounted()) {
+                            currentElement.marked = !currentElement.marked
+                            elements[i] = currentElement
+
+                            this.context.updateAppState({
+                                showSnackbar: true,
+                                snackbarConf: {
+                                    type: SUCCESS,
+                                    message: i18n.t(
+                                        currentElement.marked
+                                            ? 'Marked'
+                                            : 'Unmarked'
+                                    ),
+                                },
```

Comment:
This seems like it could be a helper function, this is quite nested already

I'd also like to suggest a more verbose message - it's VERY hard for a translator to know what `marked` / `unmarked` means since there's no context, and likewise it might not be obvious to the user.

Finally (and critically here) this use of `i18n.t` actually doesn't work - it must be called with a string literal like this `currentElement.marked ? i18n.t('Marked') : i18n.t('Unmarked')`